vars/buildExecute.groovy

```diff
@@ -102,7 +108,12 @@ void call(Map parameters = [:]) {
                     error "[${STEP_NAME}] buildTool not set and no dockerImage & dockerCommand provided."
                 }
         }
-
+        if (config.cnbBuild) {
+            if (config.buildTool in ['npm', 'maven', 'mta']) {
+                cnbBuild script: script
+            }
+            throw new AbortException("ERROR - 'cnbBuild' does not support '${config.buildTool}' as a buildTool, consider using 'kanikoExecute' instead")
+        }
```

Comment:
So lines 96-98 

```
            case ['docker', 'kaniko']:
                kanikoExecute script: script
                break
```

would be replaced with 

```suggestion
            case ['docker', 'kaniko']: //handled below
                break
```

and then below would be 

```suggestion
        if (config.cnbBuild) {
            if (config.buildTool in ['npm', 'gradle', 'maven', 'mta', 'docker']) {
                cnbBuild script: script
            }
            throw new AbortException("ERROR - 'cnbBuild' does not support '${config.buildTool}' as a buildTool, consider using 'kanikoExecute' instead")
        }
        
          if (config.kanikoExecute || (config.buildTool == 'docker' && !config.cnbBuild)) {
              kanikoExecute script: script
           }
        
```

Reply:
Do you think we need the cnbBuild bindings? For our current testing case I don't think we need them, but I'm not the expert.